gulpfile.ts

```diff
@@ -213,4 +232,231 @@ gulp.task(
+        '**/*.js',
+        ...dependencySolver.getIncludedDependencies().map((dep) => `!${dep}`)]))
+      .pipe(clean());
+  }
+
+  async function cleanEmptyDirs() {
+    await deleteEmpty(process.cwd() + '/' + output);
+  }
+
+  return gulp.series(build, filterDependencies, cleanEmptyDirs);
+}
+
+function createIsolateTasks(
+  taskNamePrefix: string,
+  input: {
+    globs: string | string[],
+    opts?: SrcOptions
+  },
+  overrideSettings: Omit<ts.Settings, 'outDir'>,
+  useCjsTransform = false,
+  tasks: {
+    name: string,
+    sourceRoots: string|string[],
+    output: string
+  }[]
+) {
+  const createdTasks: string[] = [];
+  for (const task of tasks) {
+    const taskName = `${taskNamePrefix}:${task.name}`;
+    gulp.task(
+      taskName,
+      gulpIsolateFactory(
+        {
+          sourceRoots: task.sourceRoots,
+          ...input
+        },
+        task.output,
+        overrideSettings,
+        useCjsTransform
+      )
+    );
+    createdTasks.push(taskName);
+  }
+  return createdTasks;
+}
+
+const cjsIsolateTasks = createIsolateTasks(
+  'cjs',
+  {
+    globs: 'src/**/*.ts',
+    opts: { base: 'src' }
+  },
+  {
+    target: 'ES5',
+    module: 'ES2015',
+    declaration: true
+  },
+  true,
+  [
+    {
+      name: 'stack',
+      sourceRoots: ['src/container/OtherContainer/Stack.ts'],
+      output: 'dist/isolate/cjs/stack'
+    },
+    {
+      name: 'queue',
+      sourceRoots: ['src/container/OtherContainer/Queue.ts'],
+      output: 'dist/isolate/cjs/queue'
+    },
+    {
+      name: 'priority-queue',
+      sourceRoots: ['src/container/OtherContainer/PriorityQueue.ts'],
+      output: 'dist/isolate/cjs/priority-queue'
+    },
+    {
+      name: 'vector',
+      sourceRoots: ['src/container/SequentialContainer/Vector.ts'],
+      output: 'dist/isolate/cjs/vector'
+    },
+    {
+      name: 'link-list',
+      sourceRoots: ['src/container/SequentialContainer/LinkList.ts'],
+      output: 'dist/isolate/cjs/link-list'
+    },
+    {
+      name: 'deque',
+      sourceRoots: ['src/container/SequentialContainer/Deque.ts'],
+      output: 'dist/isolate/cjs/deque'
+    },
+    {
+      name: 'ordered-set',
+      sourceRoots: ['src/container/TreeContainer/OrderedSet.ts'],
+      output: 'dist/isolate/cjs/ordered-set'
+    },
+    {
+      name: 'ordered-map',
+      sourceRoots: ['src/container/TreeContainer/OrderedMap.ts'],
+      output: 'dist/isolate/cjs/ordered-map'
+    },
+    {
+      name: 'hash-set',
+      sourceRoots: ['src/container/HashContainer/HashSet.ts'],
+      output: 'dist/isolate/cjs/hash-set'
+    },
+    {
+      name: 'hash-map',
+      sourceRoots: ['src/container/HashContainer/HashMap.ts'],
+      output: 'dist/isolate/cjs/hash-map'
+    }
+  ]
+);
+
+const esmIsolateTasks = createIsolateTasks(
+  'esm',
+  {
+    globs: 'src/**/*.ts',
+    opts: { base: 'src' }
+  },
+  {
+    target: 'ES5',
+    module: 'ES2015',
+    declaration: true
+  },
+  true,
+  [
+    {
+      name: 'stack',
```

Comment:
Make a json file for this config?

Reply:
I'll make a json config file. Should I put this on the root folder, too?

Reply:
If json file puts to `conf` folder and ts file puts to `conf` folder.

I think it can be change to this:

```typescript
const tasksArr = ['Vector', ...];
function createIsolate(module: string = 'esm', tasksArr: string[] = []) {
    return createIsolateTasks(module, ..., tasksArr.map(task => {
        return { name: task, sourceRoots: [`${module}/${task}`], output: `dist/isolate/${module}/${task}` };
    }));
} 
```